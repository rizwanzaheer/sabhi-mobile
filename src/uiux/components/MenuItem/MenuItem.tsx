import * as React from 'react'
import { TouchableHighlight } from 'react-native'
import Text from '../Text/Text'
import Container from '../Container/Container'
import Icon from '../Icon/Icon'
import { withTheme } from '../../theming'
import hexToRgba from 'hex-to-rgba'

interface MenuItem {
  /**
   * onPress function
   */
  onPress: () => void

  /**
   * Show item as active
   */
  active?: boolean

  /**
   * Provide an left icon
   */
  icon?: {
    name: string
    iconFamily: string
  }

  /**
   * Provide testID
   */
  testID?: string

  /**
   * Provided by HOC
   */
  theme: any
}

const MenuItem: React.FC<MenuItem> = props => {
  return (
    <TouchableHighlight
      style={{
        borderTopRightRadius: 20,
        borderBottomRightRadius: 20,
        backgroundColor: props.theme.colors.primary.background,
      }}
      onPress={props.onPress}
      testID={props.testID}
      accessibilityLabel={props.testID}
      underlayColor={hexToRgba(props.theme.colors.primary.accessories, 0.2)}
    >
      <Container
        viewStyle={{ borderTopRightRadius: 20, borderBottomRightRadius: 20 }}
        backgroundColor={
          props.active
            ? hexToRgba(props.theme.colors.primary.brand, 0.1)
            : hexToRgba(props.theme.colors.primary.background, 0.1)
        }
        flexDirection={'row'}
        paddingLeft={true}
        paddingTop={10}
        paddingBottom={10}
        justifyContent={'center'}
        alignItems={'center'}
      >
        {props.icon && (
          <Container>
            <Icon
              icon={props.icon}
              size={18}
              color={props.active ? props.theme.colors.primary.brand : props.theme.colors.primary.text}
            />
          </Container>
        )}
        <Container flex={1} paddingLeft={true}>
          <Text textColor={props.active && props.theme.colors.primary.brand}>{props.children}</Text>
        </Container>
      </Container>
    </TouchableHighlight>
  )
}

export default withTheme(MenuItem)
