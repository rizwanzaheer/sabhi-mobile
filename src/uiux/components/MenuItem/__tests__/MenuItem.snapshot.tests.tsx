import * as React from 'react'
import { render } from 'react-native-testing-library'
import MenuItem from '../MenuItem'

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon')

const Icon = {
  name: 'ios-settings',
  iconFamily: 'Ionicons',
}

describe('Component(snapshots): LogItem', () => {
  it('should render with required props', () => {
    const onPress = jest.fn()
    const tree = render(<MenuItem onPress={onPress}>Test Menu Item</MenuItem>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with icon', () => {
    const onPress = jest.fn()
    const tree = render(
      <MenuItem icon={Icon} onPress={onPress} testID={'TEST_MENU_ID'}>
        Test Menu Item
      </MenuItem>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render active with icon', () => {
    const onPress = jest.fn()
    const tree = render(
      <MenuItem onPress={onPress} icon={Icon} active={true}>
        Test Menu Item
      </MenuItem>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
