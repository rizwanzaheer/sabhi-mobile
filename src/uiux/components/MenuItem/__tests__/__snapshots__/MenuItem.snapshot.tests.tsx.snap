// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): LogItem should render active with icon 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomRightRadius": 20,
      "borderTopRightRadius": 20,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 120, 255, 0.1)",
        "borderBottomRightRadius": 20,
        "borderRadius": undefined,
        "borderTopRightRadius": 20,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": "center",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": 10,
        "paddingHorizontal": undefined,
        "paddingLeft": 15,
        "paddingRight": undefined,
        "paddingTop": 10,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Icon
        color="#0078ff"
        name="ios-settings"
        size={18}
      />
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": 1,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": 15,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0078ff",
            "flexWrap": "wrap",
          }
        }
      >
        Test Menu Item
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component(snapshots): LogItem should render with icon 1`] = `
<View
  accessibilityLabel="TEST_MENU_ID"
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomRightRadius": 20,
      "borderTopRightRadius": 20,
    }
  }
  testID="TEST_MENU_ID"
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.1)",
        "borderBottomRightRadius": 20,
        "borderRadius": undefined,
        "borderTopRightRadius": 20,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": "center",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": 10,
        "paddingHorizontal": undefined,
        "paddingLeft": 15,
        "paddingRight": undefined,
        "paddingTop": 10,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Icon
        color="#333333"
        name="ios-settings"
        size={18}
      />
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": 1,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": 15,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "wrap",
          }
        }
      >
        Test Menu Item
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component(snapshots): LogItem should render with required props 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomRightRadius": 20,
      "borderTopRightRadius": 20,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.1)",
        "borderBottomRightRadius": 20,
        "borderRadius": undefined,
        "borderTopRightRadius": 20,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": "center",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": 10,
        "paddingHorizontal": undefined,
        "paddingLeft": 15,
        "paddingRight": undefined,
        "paddingTop": 10,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": 1,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": 15,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "wrap",
          }
        }
      >
        Test Menu Item
      </Text>
    </View>
  </View>
</View>
`;
